packages/fxa-auth-server/lib/metrics/events.js

```diff
@@ -75,56 +87,41 @@ module.exports = (log, config) => {
      * @param {Object} [data]
      * @returns {Promise}
      */
-    emit(event, data) {
+    async emit(event, data) {
       if (!event) {
         log.error('metricsEvents.emit', { missingEvent: true });
-        return P.resolve();
+        return;
       }
 
       const request = this;
       let isFlowCompleteSignal = false;
 
-      return P.resolve()
-        .then(() => {
-          if (ACTIVITY_EVENTS.has(event)) {
```

Comment:
i wonder why this was in a separate `then`? *doesn't matter*

Reply:
I wondered the same, maybe it could have thrown?